### DISCUSSION TITLE
issue: Controllers in a form array linked to MUI Textfields do not update, unless formarray itself is altered

### DISCUSSION THREAD

**Discussion by ArneCaluwe** [react-hook-form/react-hook-form#7247] [Q&A] [answered]
### Version Number

latest (7.21.0)

### Codesandbox/Expo snack

https://codesandbox.io/s/bug-formarray-fields-not-updating-jpw4e?file=/src/AddCosts.jsx

### Steps to reproduce

* Add a new cost (click button)
* type text
* *Bug appears* 

Text can be shown by adding a new cost after typing, as 1 character will be added.

### Expected behaviour

Text should appear in the textbox labled title linked to a cost (cost being an example)

### What browsers are you seeing the problem on?

Firefox, Chrome

### Relevant log output

```shell
None
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
```tsx
<TextField
  variant="outlined"
  onChange={onChange}
  value={value ?? ""}
  label={label}
  error={!!errors}
  {...restprops} // your problem are here, it overwrite props
/>
```

**Reply by ArneCaluwe** [reply]
That's done it! Thank you!
